dot/simswap/util/reverse2original.py

```diff
@@ -100,89 +106,103 @@ def reverse2wholeimage(
     oriimg,
     pasring_model=None,
     norm=None,
-    use_mask=False,
+    use_mask=True,
     use_gpu=True,
+    use_cam=True,
 ):
 
-    target_image_list = []
-    img_mask_list = []
+    device = torch.device("cuda" if use_gpu else "cpu")
     if use_mask:
-        if use_gpu:
-            smooth_mask = SoftErosion(
-                kernel_size=17, threshold=0.9, iterations=7
-            ).cuda()
-        else:
-            smooth_mask = SoftErosion(kernel_size=17, threshold=0.9, iterations=7).cpu()
+        smooth_mask = SoftErosion(kernel_size=17, threshold=0.9, iterations=7).to(
+            device
```

Comment:
can you make the `device` as a variable to be passed in the constructor instead?

Reply:
I think using `device` like this is the best way. 
Do you think using the `device` variable in the `forward` function is better?